react/features/app/actions.js

```diff
@@ -31,176 +31,70 @@ declare var APP: Object;
-
-    /**
-     * Notifies that an attempt to load a configuration has completed. Due to
-     * the asynchronous nature of the loading, the specified {@code config} may
-     * or may not be required by the time the notification arrives.
-     *
-     * @param {string|undefined} error - If the loading has failed, the error
-     * detailing the cause of the failure.
-     * @param {Object|undefined} config - If the loading has succeeded, the
-     * loaded configuration.
-     * @returns {void}
-     */
-    function loadConfigSettled(error, config) {
-        // Due to the asynchronous nature of the loading, the specified config
-        // may or may not be required by the time the notification arrives. If
-        // we receive the config for a location we are no longer interested in,
-        // "ignore" it - deliver it to the external API, for example, but do not
-        // proceed with the appNavigate procedure/process.
-        if (getState()['features/base/config'].locationURL === locationURL) {
-            dispatch(setLocationURL(locationURL));
-            dispatch(setConfig(config));
-        } else {
-            // eslint-disable-next-line no-param-reassign
-            error || (error = new Error('Config no longer needed!'));
-
-            // XXX The failure could be, for example, because of a
-            // certificate-related error. In which case the connection will fail
-            // later in Strophe anyway.
-            dispatch(loadConfigError(error, locationURL));
-
-            throw error;
-        }
-    }
-}
-
-/**
- * Triggers an in-app navigation to a specific or undefined location (URI).
- *
- * @param {Dispatch} dispatch - The redux {@code dispatch} function.
- * @param {Function} getState - The redux function that gets/retrieves the redux
- * state.
- * @param {Object} location - The location (URI) to navigate to. The value may
- * be undefined.
- * @private
- * @returns {void}
- */
-function _appNavigateToOptionalLocation(
-        dispatch: Dispatch<any>, getState: Function,
-        location: Object) {
-    // If the specified location (URI) does not identify a host, use the app's
-    // default.
-    if (!location || !location.host) {
-        const defaultLocation = parseURIString(getDefaultURL(getState));
-
-        if (location) {
-            location.host = defaultLocation.host;
-
-            // FIXME Turn location's host, hostname, and port properties into
-            // setters in order to reduce the risks of inconsistent state.
-            location.hostname = defaultLocation.hostname;
-            location.pathname
-                = defaultLocation.pathname + location.pathname.substr(1);
-            location.port = defaultLocation.port;
-            location.protocol = defaultLocation.protocol;
-        } else {
-            // eslint-disable-next-line no-param-reassign
-            location = defaultLocation;
+    return async(dispatch: Dispatch<any>, getState: Function) => {
```

Comment:
style: space after async

Reply:
Nope, eslint doesn't let me

Reply:
The we need a new rule or something. `async` needs a space, otherwise `async function foo() { ... }` cannot be written.

Reply:
No, that would work, this is a separate eslint rule, see PR: https://github.com/jitsi/eslint-config-jitsi/pull/10